crates/outline_panel/src/outline_panel.rs

```diff
@@ -4718,6 +4919,45 @@ impl OutlinePanel {
             })
             .collect()
     }
+
+    fn has_outline_children(&self, outline: &OutlineEntryOutline, _cx: &App) -> bool {
+        // Check the actual excerpt outlines to see if this outline has children
+        if let Some(excerpts) = self.excerpts.get(&outline.buffer_id) {
+            if let Some(excerpt) = excerpts.get(&outline.excerpt_id) {
+                let outlines: Vec<_> = excerpt.iter_outlines().collect();
+
+                // Find the current outline
+                for (i, o) in outlines.iter().enumerate() {
```

Comment:
We can skip all outlines before the current one, similarly we can skip the tail if never hit the `if` condition, right?